IotcRestApi/regulator_draft.md

```diff
@@ -0,0 +1,145 @@
+# API
+Bude se jednat o REST API, bude k dispozici Swagger.
+
+# Zabezpečení
+Komunikace bude zabezpečena bearer tokenem v hlavičce, informace budou doplněny.
+
+Většina endpointů přijímajících data v těle requestu vyžaduje položku `requestId`. Jedná se o jednoznačný identifikátor requestu, který generuje partner (jedná se o GUID) a slouží pro zajištění idempotence, čímž je docíleno toho, že akce je provedena jen jednou pokud dojde o opakovanému zavolání v určitém časovém intervalu.
+
+# Chybové odpovědi
+Chyby 5xx jsou způsobené chybou serveru a neměli by nikdy nastat. Pokud server vratí tuto chybu tak
+by měl být okamžitě informován zástupce z firmy Netlia.
+
+Chyby 4xx jsou vráceny, pokud klient provedl neplatný/nevalidní request. Vysvětlení stavových kódu je možné najít
+v HTTP dokumentaci - https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#client_error_responses. Podporované stavové kódy budou doplněny.
+
+V některých případech 4xx error obsahuje body s dalšimy informacemi o chybě. Body má následující formát:
+
+```yaml
+{
+    "errors" : [
+        {
+            "errorCode":int
+            "description":string
+            "additionalInformation":string
+            "traceId":string
+        }
+    ]
+}
+```
+
+# Zařízení a podporované endpointy
+
+## Teplotní regulátor
+
+### POST api/temperature-regulator/{DeviceId}/mode
```

Comment:
Tohle uz je PUT jelkoz mame requestId a je to idempotentni.

Reply:
ok